src/python/pants/init/engine_initializer.py

```diff
@@ -335,14 +336,26 @@ def setup_legacy_graph_extended(
                                    exclude_target_regexps=exclude_target_regexps,
                                    subproject_roots=subproject_roots)
 
+    @rule(GlobMatchErrorBehavior, [])
+    def glob_match_error_behavior_singleton():
+      return glob_match_error_behavior or GlobMatchErrorBehavior.ignore
```

Comment:
I was going to say it might be interesting to have a more concise wrapper for these expressions, then realized that's exactly what we're removing in this diff. I'm definitely 100% ok with using `@rule` here there and everywhere -- something special is happening (we are creating a rule!) and it makes perfect sense to require that it be specially annotated.

Reply:
Yea. And I have a sneaking suspicion that singletons are generally anti-patterns anyway.

Certainly many of the things that are currently singletons probably should not be in the long term: instead, they should be constructed based on options parsing... and the inputs to options parsing would be Params.